code/src/pages/products/ProductsPage.tsx

```diff
@@ -1,23 +1,40 @@
+import { useState } from "react";
 import AppTypography from "@/components/app-typography/AppTypography";
 import AppBox from "@/components/app-box/AppBox";
 import ProductCard from "@/components/product-card/ProductCard";
 import PageWrapper from "@/layouts/page-wrapper/PageWrapper";
+import AppDropdown from "@/components/app-dropdown/AppDropdown";
+
+import { sortOptions } from "@/pages/products/ProductsPage.constants";
+import { Product } from "@/types/product.types";
 import { useGetProductsQuery } from "@/store/api/productsApi";
 
 import "@/pages/products/ProductsPage.scss";
 
 const ProductsPage = () => {
-  const { data: products, isLoading } = useGetProductsQuery();
+  const [sortOption, setSortOption] = useState("");
+  const { data: products, isLoading: productsLoading } =
+    useGetProductsQuery(sortOption);
```

Comment:
Make `sortOption` optional

Reply:
> Make `sortOption` optional

What do you mean ?
I can just pass empty variable here. It doesn't matter

Reply:
Please don't be the doesn't matter person. Despite the product is mostly for studing purposes, we should try to make it as good as we can. Moreover, it is not hard to tweek a type and our code will become a little better. All best things consists of a bunch of small things we can do ourselves

Reply:
> Please don't be the doesn't matter person. Despite the product is mostly for studing purposes, we should try to make it as good as we can. Moreover, it is not hard to tweek a type and our code will become a little better. All best things consists of a bunch of small things we can do ourselves

Give some suggesting to review please and not to replace `AppBox` to `AppTypography` and change mock data for tests

And give me answer for suggestion about `sortOptional` optional. I don't understand what do you mean

Reply:
Same here, we can try to use object for example passed to `useGetProductsQuery`:

```jsx
{
    sortStrategy: "newest",
    language: "en",
    page: 0,
    ...
}
```

I think your hook will be more configurable from backend side and it will be better to use object to apply filters, sorting, language and so on

I took url from backend as a reference
http://localhost:8080/retail/v1/products?lang=en&page=0&size=8&sort=price,asc